regression/cbmc/union11/union_list.c

```diff
@@ -0,0 +1,30 @@
+#include <assert.h>
+
+struct list_item
+{
+  struct list_item *previous;
+};
+
+struct List
+{
+  unsigned something;
+  struct list_item *index;
+  struct list_item end;
+};
+
+union U
+{
+  struct List my_list;
```

Comment:
Comment inline about whether the single-item union is important or not, since that's pretty unusual afaik